### PR TITLE
Bug 1907309: Add APF StorageVersionMigration manifest

### PR DESCRIPTION
**Pull request by ihcsim**
This PR updates the `StaticResourceController` to use a new
`StorageVersionMigration` manifest to manage APF resource storage version
migration.

This PR is dependent on https://github.com/openshift/library-go/pull/1091.

Signed-off-by: Ivan Sim <isim@redhat.com>

### CODE REVIEW

**Comment by deads2k** [comment]
This matches expectations, but I think you need a resourcread.

**Comment by deads2k** [comment]
if this worked, release the hold on the library-go PR

File: bindata/v4.1.0/kube-apiserver/storage-version-migration.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+spec:
+  resource:
+    group: flowcontrol.apiserver.k8s.io
+    version: v1beta1
+    resource: flowschemas
+---
+apiVersion: migration.k8s.io/v1alpha1
+kind: StorageVersionMigration
+metadata:
+  name: flowcontrol-prioritylevel-storage-version-migration
```
Comment (deads2k) [inline bindata/v4.1.0/kube-apiserver/storage-version-migration.yaml:14]: looks like you need two files.  But otherwise I think it worked.

**Comment by ihcsim** [comment]
Updated go.mod to vendor https://github.com/openshift/library-go/commit/abe9feaede8b357ced33bf3ba2606434f6235700.

Retested storage version migration with a 4.7 -> 4.8 cluster upgrade:

```sh
$ oc -n openshift-kube-apiserver-operator logs kube-apiserver-operator-64f8bb6d4d-6fkkn | grep "flowcontrol"
I0607 18:24:50.328181       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-kube-apiserver-operator", Name:"kube-apiserver-operator", UID:"3531372e-c99a-4e55-b94f-94a6c6594cae", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'StorageVersionMigrationCreated' Created StorageVersionMigration.migration.k8s.io/flowcontrol-flowschema-storage-version-migration because it was missing
I0607 18:24:50.335725       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-kube-apiserver-operator", Name:"kube-apiserver-operator", UID:"3531372e-c99a-4e55-b94f-94a6c6594cae", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'StorageVersionMigrationCreated' Created StorageVersionMigration.migration.k8s.io/flowcontrol-prioritylevel-storage-version-migration because it was missing

$ oc -n openshift-kube-storage-version-migrator logs migrator-68b46b456d-4tf4x | grep flowcontrol                     
I0607 18:24:50.966161       1 kubemigrator.go:110] flowcontrol-flowschema-storage-version-migration: migration running
I0607 18:24:51.146760       1 kubemigrator.go:127] flowcontrol-flowschema-storage-version-migration: migration succeeded
I0607 18:24:52.156875       1 kubemigrator.go:110] flowcontrol-prioritylevel-storage-version-migration: migration running
I0607 18:24:52.256543       1 kubemigrator.go:127] flowcontrol-prioritylevel-storage-version-migration: migration succeeded

```

File: vendor/github.com/openshift/library-go/pkg/controller/factory/eventfilters.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+func NamesFilter(names ...string) EventFilterFunc {
+	nameSet := sets.NewString(names...)
+	return func(obj interface{}) bool {
+		metaObj, ok := obj.(metav1.ObjectMetaAccessor)
+		if !ok {
+			return false
+		}
+		return nameSet.Has(metaObj.GetObjectMeta().GetName())
+	}
+}
```
Comment (ihcsim) [inline vendor/github.com/openshift/library-go/pkg/controller/factory/eventfilters.go:26]: This was pulled as part of bumping library-go. It isn't part of my change.

**Pull request merged**